### PR TITLE
Repovation weekly PR

### PR DESCRIPTION
**Pull request by tamtamchik**
Updates related to Lido core repository "repovation". Shipped to master weekly.

## Context

- https://github.com/lidofinance/core/pull/200
- https://github.com/lidofinance/core/pull/166
- https://github.com/lidofinance/core/pull/136
- https://github.com/lidofinance/core/pull/198
- https://github.com/lidofinance/core/pull/202
- https://github.com/lidofinance/core/pull/203 + https://github.com/lidofinance/core/pull/205
- https://github.com/lidofinance/core/pull/206
- https://github.com/lidofinance/core/pull/207
- Dependencies updates

### CODE REVIEW

**Comment by tamtamchik** [comment]
@lidofinance/review-gh-workflows sorry for the size of the PR, but here only 2 files with minor changes, don't think it'll take too long to review.

**Comment by rnmsslido** [comment]
LGTM for the changes in `.github/workflows`. Not reviewing other parts of the PR as they are outside the scope of `review-gh-workflows` group.

**Review by rnmsslido — APPROVED** [review]

**Review by mymphe — APPROVED** [review]

**Review by TheDZhon — CHANGES_REQUESTED** [review]
What a titanic work made here :fire: 

I asked a few questions here and there; please consider them favorably with an attitude to make the solution more future-proof :+1:

File: scripts/dao-upgrade.sh
[the remark this answers is not in the document]
Comment (tamtamchik) [reply scripts/dao-upgrade.sh:1]: I think we can, but I would prefer to create a separate task for this along with the refactoring of the deployment steps.

**Review by tamtamchik — COMMENTED** [review]
Thank you for your feedback! ❤️  
I have addressed some of the raised issues in PR #207 / added my comments to continue the discussion here.

**Review by TheDZhon — APPROVED** [review]
Gotta be big :muscle:

**Pull request merged**